saleor/order/base_calculations.py

```diff
@@ -95,11 +95,10 @@ def propagate_order_discount_on_order_prices(
     """
     base_subtotal = base_order_subtotal(order, lines)
     subtotal = base_subtotal
-    shipping_price = order.undiscounted_base_shipping_price
+    shipping_price = order.base_shipping_price
```

Comment:
Cannot add comment in line `139`, but there `order.undiscounted_base_shipping_price` should be used. Otherwise, you will apply the manual discount on the discounted shipping price.
Also, we need a test for that